os/fs/semaphore/sem_open.c

```diff
@@ -261,6 +264,10 @@ FAR sem_t *sem_open(FAR const char *name, int oflags, ...)
 			/* Return a reference to the semaphore */
 
 			sem = &nsem->ns_sem;
+
+#ifdef CONFIG_BINARY_MANAGER
+			binary_manager_register_resource(BIN_RESOURCE_NAMEDSEM, name);
```

Comment:
like a above code, you are doing kernel operation here, so you have to consider errno.
I think Void is not acceptable as a function type..